test/e2e/knative_serving_test.go

```diff
@@ -0,0 +1,163 @@
+package e2e
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/openshift-knative/serverless-operator/test"
+)
+
+const (
+	knativeServing    = "knative-serving"
+	testNamespace     = "serverless-tests"
+	image             = "gcr.io/knative-samples/helloworld-go"
+	helloworldService = "helloworld-go"
+)
+
+func TestKnativeServing(t *testing.T) {
+	contexts := test.Setup(t)
+	adminCtx := contexts[0]
+
+	defer test.CleanupAll(contexts)
+	test.CleanupOnInterrupt(t, func() { test.CleanupAll(contexts) })
+
+	t.Run("create subscription and wait for CSV to succeed", func(t *testing.T) {
+		_, err := test.WithOperatorReady(adminCtx, "serverless-operator-subscription")
+		if err != nil {
+			t.Fatal("Failed", err)
+		}
+	})
+
+	t.Run("deploy knativeserving cr and wait for it to be ready", func(t *testing.T) {
+		_, err := test.WithKnativeServingReady(adminCtx, knativeServing, knativeServing)
+		if err != nil {
+			t.Fatal("Failed to deploy KnativeServing", err)
+		}
+	})
+
+	t.Run("deploy knative service using kubeadmin", func(t *testing.T) {
+		_, err := test.WithServiceReady(adminCtx, helloworldService, testNamespace, image)
+		if err != nil {
+			t.Fatal("Knative Service not ready", err)
+		}
+	})
+
+	t.Run("user permissions", func(t *testing.T) {
+		testUserPermissions(t, contexts)
+	})
+
+	t.Run("undeploy serverless operator and check dependent operators removed", func(t *testing.T) {
+		adminCtx.Cleanup()
+		err := test.WaitForOperatorDepsDeleted(adminCtx)
+		if err != nil {
+			t.Fatalf("Operators still running: %v", err)
+		}
+	})
+}
+
+func testUserPermissions(t *testing.T, contexts []*test.Context) {
+	editCtx := contexts[1]
+	viewCtx := contexts[2]
+	tests := []struct {
+		name          string
+		userRole      string
+		userContext   *test.Context
+		operationName string
+		operation     func(context *test.Context) error
```

Comment:
`operationName` and `operation` seem entirely redundant. Maybe infer the operation from the string in the `Run` block below?

Reply:
I can just remove the operationName, it's always clear what the operation is - either from the test name or from the error being thrown.